File: src/main/java/com/the9grounds/aeadditions/api/AEAApi.java
package com.the9grounds.aeadditions.api;

public class AEAApi {

	public static IAEAdditionsAPI instance() {
		if (instance == null) {
			try {
				instance = (IAEAdditionsAPI) Class
					.forName("com.the9grounds.aeadditions.AEAdditionsApi")
					.getField("instance").get(null);
			} catch (Exception e) {
			}
		}
		return instance;
	}

	private static IAEAdditionsAPI instance = null;

}

File: src/main/kotlin/com/the9grounds/aeadditions/api/AEAApi.kt
package com.the9grounds.aeadditions.api

import appeng.api.stacks.AEKeyType

object AEAApi {
    val superCellTypes = mutableListOf(
        AEKeyType.items(),
        AEKeyType.fluids()
    )

    fun registerTypeForSuperCell(keyType: AEKeyType) {
        superCellTypes.add(keyType)
    }
}